include/zephyr/storage/disk_access.h

```diff
@@ -100,6 +100,18 @@ int disk_access_read(const char *pdrv, uint8_t *data_buf,
 int disk_access_write(const char *pdrv, const uint8_t *data_buf,
 		      uint32_t start_sector, uint32_t num_sector);
 
+/**
+ * @brief erase data from disk
+ *
+ * This command will result in the specified sectors containing either
+ * 0x00 or 0xFF bytes.
```

Comment:
IMO, the API should not claim or enforce these values. There are good reasons to wipe a disk with random data, even if some flash-based devices do not support it.

Reply:
These are not enforced by us. In this case the MMC controller logic decides. We can not. You can only query the device for what value will be placed on silicon once erased.
This also differs from scrambling the disk, as you have described, because it actually does the "Flash" type erase where underlying storage solution is reset to non-written state; main intent of erase on MMC is to speed up write operations, on released sectors, this is exactly reasoning behind original idea behind Flash type devices, as evolution from EEPROM (this and cost of transistor).
What controller does it actually writes opposite bit value, on selected range, to the non-written state, and the resets all bits in range to non-written state.

Reply:
>These are not enforced by us. 

They are if the text above is part of the API description and the test depends on these values (0x00, 0xFF) regardless of the device type.

Reply:
Ah, right with ramdisk we actually can decide. Nevertheless I would rather leave the meaning of erase as is, with no expectation of scrambling data and provide "scramble" variant of function that would do that.

Reply:
Why wouldn't we enforce this? I think wiping a disk with random data is a valid use case, but a different one from what we have here. Perhaps wiping a disk with random data is a "secure erase"?

Reply:
Because this misses the entire point of erase in the first place. The MMC erase/trim is intended for improved write times, it is not for "scrambling data". The controller uses value that is available for it, depending on the technology.

Both MMC operations, trim and erase, already have a secure variant, which again is something different than scrambling data - it just changes the erase/trim schedule.

> You also need to take a look at the MMC Product Standard JESD84-A441 from JEDEC (https://www.jedec.org/system/files/docs/JESD84-A441.pdf) , as it describes the TRIM and ERASE operations and their secure variants.